### PR TITLE
feat: Updated BGSMod and TESObjectWEAP headers

### PR DESCRIPTION
**Pull request by Zzyxz**
TESObjectWEAP

- New Classes Added: Introduced several classes related to weapon data handling, including:
- WeaponAmmoData, WeaponDamageData, WeaponFiringData, WeaponDataFlags, WeaponDataGeneral, WeapDataMelee, WeaponDataPower, WeapDataReload, WeapDataTurret, WeapDataVariableRange, WeapDataVisuals, WeapDataShipWeapon, WeaponDataAudio, WeaponDataCharge, and WeaponDataAim.

- Class TESObjectWEAPInstanceData Updated: Modified to include pointers to these new weapon data classes.

- Static Assertions Added: Included multiple static_assert statements to validate offsets within the new classes, ensuring proper memory alignment for each member variable.

BGSMod

- New Class (`BGSAddonNode`): Added to the `RE` namespace.
- Property Structure: Introduced a new `Property` structure inside `Data`, supporting different value types (`bool`, `float`, `int`). Used for OMOD Properties.

### CODE REVIEW

**Comment by qudix** [comment]
you should change all uses of `stl::enumeration<T, E>` to `REX::EnumSet<T, E>`, under the hood `stl::enumeration` is just an alias for `EnumSet` for compatibility reasons

File: include/RE/B/BGSMod.h [hunk 1]
```diff
@@ -45,72 +48,114 @@ namespace RE::BGSMod
 
 				struct Include
 				{
-					BGSMod::Attachment::Mod*                 mod;
-					std::uint8_t                             level;
-					REX::EnumSet<IncludesFlag, std::uint8_t> flags;
+					RE::BGSMod::Attachment::Mod* mod;
```
Comment (qudix) [inline include/RE/B/BGSMod.h:51]:
```suggestion
					BGSMod::Attachment::Mod* mod;
```

File: include/RE/B/BGSMod.h [hunk 2]
```diff
@@ -45,72 +48,114 @@ namespace RE::BGSMod
...
-		{
-		public:
-			SF_RTTI_VTABLE(BGSMod__Template__Item);
-
-			~Item() override;  // 00
-		};
-
-		class __declspec(novtable) Items :
-			public BaseFormComponent  // 00
+		class Items : public BaseFormComponent
```
Comment (qudix) [inline include/RE/B/BGSMod.h:138]: Is your fork out of date, or did you remove these for a reason?

File: include/RE/B/BGSMod.h [hunk 3]
```diff
@@ -45,72 +48,114 @@ namespace RE::BGSMod
...
-			BSFixedString   unk18;  // 18
+			BSTArray<void*> unk08; // 08
+			BSFixedString   unk18; // 18
 		};
+
 		static_assert(sizeof(Items) == 0x20);
-	}
-}
+	} // namespace Template
+} // namespace RE::BGSMod
```
Comment (qudix) [inline include/RE/B/BGSMod.h:161]:
```suggestion
	}
}
```

File: include/RE/B/BGSMod.h (hunk 2 above)
Comment (Zzyxz) [reply include/RE/B/BGSMod.h:138]: Could be, yes. It's my very first time trying this ^^ I wanted to contribute at least something

**Pull request merged**